src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -77,6 +83,10 @@ private void bindHttpServer(Future<Void> startFuture) {
                                         .get(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)
                         );
 
+                        if (this.bridgeConfig.getHttpConfig().getConsumerTimeout() > -1) {
+                            startInactiveConsumerDeletionTimer(this.bridgeConfig.getHttpConfig().getConsumerTimeout());
```

Comment:
I think we have a problem with this condition. If the user sets 0 as timeout we'll have consumers removed after the first timer tick (1 sec).
The condition in the timer handler:

`if (item.getValue() + timeout < System.currentTimeMillis())`

will always be true (the consumer timestamp was put in the map before the first tick timer ever and because timeout is 0, it will always be less than the current time). So we are goint to remove the consumer after max one sec (or even less, it depends on the time alignment with the timer when the consumer is created and the timestamp is put into the map). I am still thinking that we should not allow negavite values (as -1) and using 0 for disabling while missing the `http.timeout` property in the configuration means using the default value (as for the other properties). @stanlyDoge @tombentley any more thoughts?

Reply:
I'm not sure that helps, because the user can still set the timeout as, say, 1 with the terrible consequence that the consumer gets removed after 1 second or less. 

We could say that the config is either a strictly positive integer or the word `off`, which avoids any confusion around having a special value.

Also. I do hope the config will be `http.timeoutMs` or (even better) `http.timeoutSeconds` (since ms precision just isn't warranted), rather than the units-ambiguous `http.timeout`.

Reply:
> I'm not sure that helps, because the user can still set the timeout as, say, 1 with the terrible consequence that the consumer gets removed after 1 second or less.

In this case it's a user mistake, isn't it? You should expect that if you set timeout to 1 sec.

> We could say that the config is either a strictly positive integer or the word off, which avoids any confusion around having a special value.

Tbh I don't like the Kafka way to have `acks` which can be 0, 1 or "all" (mixing numbers and words). I would stay with a strict positive number or 0 to disable ...

Finally, I agree on the parameter name. I would go for `timeoutSeconds`

Reply:
I just don't think that saying timeout=0 means no timeout is at all intuitive. `-1` (and `off` even more so) is more obviously not a normal value. By all means make 0 an error and bail out on start up, but giving it an unintuitive meaning is just going to confuse people.

Reply:
At this point I think we need one more comment from a maintainer, @scholzj your opinion is important here :-)